pkg/compactor/visit_marker_test.go

```diff
@@ -0,0 +1,277 @@
+package compactor
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/oklog/ulid"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+	"github.com/thanos-io/objstore"
+	"github.com/thanos-io/thanos/pkg/compact"
+
+	cortex_testutil "github.com/cortexproject/cortex/pkg/storage/tsdb/testutil"
+)
+
+func TestMarkPending(t *testing.T) {
+	ctx := context.Background()
+	dummyCounter := prometheus.NewCounter(prometheus.CounterOpts{})
+	bkt, _ := cortex_testutil.PrepareFilesystemBucket(t)
+	logger := log.NewNopLogger()
+
+	ownerIdentifier := "test-owner"
+	testVisitMarker := NewTestVisitMarker(ownerIdentifier)
+
+	visitMarkerManager := NewVisitMarkerManager(objstore.WithNoopInstr(bkt), logger, ownerIdentifier, testVisitMarker, dummyCounter, dummyCounter)
+	visitMarkerManager.MarkPending(ctx)
+
+	require.Equal(t, Pending, testVisitMarker.Status)
+
+	visitMarkerFromFile := &TestVisitMarker{}
+	err := visitMarkerManager.ReadVisitMarker(ctx, visitMarkerFromFile)
+	require.NoError(t, err)
+	require.Equal(t, Pending, visitMarkerFromFile.Status)
+}
+
+func TestMarkInProgress(t *testing.T) {
+	ctx := context.Background()
+	dummyCounter := prometheus.NewCounter(prometheus.CounterOpts{})
+	bkt, _ := cortex_testutil.PrepareFilesystemBucket(t)
+	logger := log.NewNopLogger()
+
+	ownerIdentifier := "test-owner"
+	testVisitMarker := NewTestVisitMarker(ownerIdentifier)
+
+	visitMarkerManager := NewVisitMarkerManager(objstore.WithNoopInstr(bkt), logger, ownerIdentifier, testVisitMarker, dummyCounter, dummyCounter)
+	visitMarkerManager.MarkInProgress(ctx)
+
+	require.Equal(t, InProgress, testVisitMarker.Status)
+
+	visitMarkerFromFile := &TestVisitMarker{}
+	err := visitMarkerManager.ReadVisitMarker(ctx, visitMarkerFromFile)
+	require.NoError(t, err)
+	require.Equal(t, InProgress, visitMarkerFromFile.Status)
+}
+
+func TestMarkCompleted(t *testing.T) {
+	ctx := context.Background()
+	dummyCounter := prometheus.NewCounter(prometheus.CounterOpts{})
+	bkt, _ := cortex_testutil.PrepareFilesystemBucket(t)
+	logger := log.NewNopLogger()
+
+	ownerIdentifier := "test-owner"
+	testVisitMarker := NewTestVisitMarker(ownerIdentifier)
+
+	visitMarkerManager := NewVisitMarkerManager(objstore.WithNoopInstr(bkt), logger, ownerIdentifier, testVisitMarker, dummyCounter, dummyCounter)
+	visitMarkerManager.MarkCompleted(ctx)
+
+	require.Equal(t, Completed, testVisitMarker.Status)
+
+	visitMarkerFromFile := &TestVisitMarker{}
+	err := visitMarkerManager.ReadVisitMarker(ctx, visitMarkerFromFile)
+	require.NoError(t, err)
+	require.Equal(t, Completed, visitMarkerFromFile.Status)
+}
+
+func TestReloadVisitMarker(t *testing.T) {
```

Comment:
I see we removed `ReloadVisitMarker` method already. Can we remove this test case?